caravel/assets/javascripts/explore/components/DisplayQueryButton.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { PropTypes } from 'react';
+import ModalTrigger from './../../components/ModalTrigger';
+
+const propTypes = {
+  slice: PropTypes.object.isRequired,
+};
+
+export default class DisplayQueryButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      viewSqlQuery: '',
+    };
+    this.beforeOpen = this.beforeOpen.bind(this);
+  }
+
+  beforeOpen() {
+    this.setState({
+      viewSqlQuery: this.props.slice.viewSqlQuery,
```

Comment:
this.props.slice gets updated after everything is mounted, so before we open the modal we need to set the state with the updated slice data. i should add a todo, to fix this once caravel.js is refactored to better work with the react structure/lifecycle.